docs/events.md

```diff
@@ -0,0 +1,175 @@
+  if both players are ready, we emit a `game:your turn` to one
+  of the player and we clear the board
+---
+
+### game:join
+- name: `game:join`
+- args: id(String), player(String)
+- source: **client**
+- destination: multiple
+- reaction: `game:create`(service) | `game:joined`(client) | `game:full`(client)
+- description:
+
+    Triggered when a user wants to join a game  
+
+    if the game does not exists, we fire a `game:create`
+    to create a game with the id
+
+    else if the game exists but is full,
+    we emit a `game:full` to the client
+
+    else we insert the player to the game,
+    we emit `game:joined` to the client
+    and we emit a `game:start` as a service
+---
+
+## Service events
+> These events can be emitted from any services
+
+### game:create
+- name: `game:create`
+- args: id(String), Player(String)
+- source: **service**
+- destination: client
+- reaction: `game:joined`(client)
+- description:
+
+    Triggered when a client has emitted a `game:join`
+    but the game does not exists yet.
+    We create the game, put the player in and emit to
+    the client a `game:join`
+---
+
+### game:joined
+- name: `game:join`
+- args: board(Array), me(Object), him(String)
+- source: **service**
+- destination: client
+- reaction: no
+- description:
+
+    Triggered when the user is connected to a game
+    He can now setup the `game:your turn` event
+    to play when it is his turn
+---
+
+### game:full
+- name: `game:full`
+- args: errorMessage(String)
+- source: **service**
+- destination: client
+- reaction: no-reaction
+- description:
+    
+    Triggered when a user has asked to join a
+    game and the game is full
+---
+
+### game:your turn
+- name: `game:your turn`
+- args: board(Array)
+- source: **service**
+- destination: client
+- reaction: no-reaction
+- description:
+
+  Triggered when the client have to put a piece on the
+  board. We can now setup the `game: put a piece`
+---
+
+### game:draw
+- name `game:draw`
+- args: no
+- source: **service**
+- destination: client
+- reaction: no-reaction
+- description:
+
+  Triggered when the board is full, both players have to
+  emit a `game:replay` to restart the game
+---
+
+### game:end
+- name: `game:end`
+- args: winner(Object)
+- source: **service**
+- destination: client
+- reaction: no-reaction
+- description:
+
+  Triggered when the game is ended because we have a winner.
+  We can now waiting for a `game:replay` from both players
+---
+
+### game:cannot put a piece
+- name: `game:cannot put a piece`
+- args: no
+- source: **service**
+- destination: client
+- reaction: no-reaction
+- description:
+
+  Triggered when the client has tried to put a piece but she cannot
+  be put where he ask for some reason
+---
+
+### game:leaving
+- name: `game:leaving`
+- args: no
+- source: **service**
+- destination: client
+- reaction: no-reaction
+- description:
+
+  Triggered when the other player has left the game
+---
+
+### game:leave
```

Comment:
useful ? if a user leave due to a page reload, other player will receive a 'end' event with may be an explanation